src/moneypenny/handlers/external/agent.py

```diff
@@ -68,20 +72,21 @@ async def commission_agent(request: web.Request) -> web.Response:
         ),
     )
     moneypenny = request.config_dict["moneypenny"]
-    username = body["username"]
+    username = body["token"]["data"]["uid"]
     await moneypenny.execute_order(verb="post", dossier=body)
```

Comment:
When you're passing in the name of the function to call as a string argument to a method, it feels like something has gone wrong. Why isn't the internal API just `provision()` and `deprovision()` methods?

Reply:
Well, we now have "commission" and "retire" but they're still calling a single function back in moneypenny, because what moneypenny does to execute the function is to construct a Pod with a sequence of containers, and execute them in order, no matter what the function is.